openedx/core/djangoapps/django_comment_common/comment_client/thread.py

```diff
@@ -171,17 +172,14 @@ def _retrieve(self, *args, **kwargs):
             'merge_question_type_responses': kwargs.get('merge_question_type_responses', False)
         }
         request_params = utils.strip_none(request_params)
-        course_id = kwargs.get("course_id")
-        if not course_id:
-            course_id = forum_api.get_course_id_by_thread(self.id)
-        course_key = utils.get_course_key(course_id)
-        if is_forum_v2_enabled(course_key):
+        use_forumv2, course_id = is_forum_v2_enabled_for_thread(self.id)
```

Comment:
can we use same if-else block that we used above in models.py.
```
if course_id:
    use_forumv2 = is_forum_v2_enabled(course_id)
else:
    use_forumv2, course_id = is_forum_v2_enabled_for_thread(self.id)
```

Reply:
Yes.